src_code/src/controllers/services/Authentication/SendVerificationCode.js

```diff
@@ -0,0 +1,127 @@
+        }
+
+        /** If type of isNewAccount is not a bool, return bad request */
+        if(typeof req.isNewAccount !== "boolean"){
+            return responseBuilder.BadRequest(res, "Invalid type in request body.");
+        }
+
+        /** If the request is for newAccount (Sign Up), help client check emailAddress and studentId existence */
+        if(req.isNewAccount === true) {
+            const isEmailExist = await dbHelper.GetUserInfoByEmailAddress(db, emailAddress);
+            if(isEmailExist && typeof isEmailExist === "string"){
+                return responseBuilder.BadRequest(res, isEmailExist);
+            }
+            if(isEmailExist) {
+                return responseBuilder.BadRequest(res, "Email address is already used.");
+            }
+            if(!numPattern.test(req.studentId)) {
+                return responseBuilder.BadRequest(res, "Invalid student ID.");
+            }
+            const isStudentIdExist = await dbHelper.GetUserInfoByStudentId(db, req.studentId.trim());
+            /** Check if there is error while retrieving info by student id */
+            if(isStudentIdExist && typeof isStudentIdExist === "string") {
+                return responseBuilder.BadRequest(res, isStudentIdExist);
+            }
+            /** If there is already an user with the sign up student id, return bad request */
+            if(isStudentIdExist) {
+                return responseBuilder.BadRequest(res, "Student ID is already used.");
+            }
+        }
+        
+        /** Validate ensure firstName and lastName should be less than 25 characters (Database restriction) */
+        if(firstName.length > 25 || lastName.length > 25) {
+            return responseBuilder.BadRequest(res, "First name and last name must be less than 25 characters.");
+        }
+
+        /** Ensure email is always @spu.edu domain */
+        if(!emailPattern.test(emailAddress.trim())) {
+            return responseBuilder.BadRequest(res, "Invalid email address. Please try again.");
+        }
+
+        /** Ensure that the verification code is valid. Should be four in length and a string type */
+        if(verificationCode.length !== 4 || !/^[0-9]+$/.test(verificationCode)) {
+            /** If verificationCode is not a string, then return bad request */
+            if(typeof verificationCode !== "string") {
+                return responseBuilder.BadRequest(res, "Invalid type of verification code. Please try again.");
+            } 
+            /** Return bad request for all bad format verification code */
+            return responseBuilder.BadRequest(res, "Invalid verification code. Please try again.");
+        }
+
+        /** Return null to indicate pass the validation */
+        return null;
+    } catch(error) {
+        /** If there is unexpected error, we catch it, log and return error */
+        console.log("ERROR: There is an error occurred while validating sending verification code: ", error);
```

Comment:
Another testing console.log?

Reply:
It is for AWS Cloudwatch Log to log bug, that will not be appeared on the frontend.